server/filestore.go

```diff
@@ -2266,7 +2265,7 @@ func (mb *msgBlock) firstMatchingMulti(sl *Sublist, start uint64, sm *StoreMsg)
 		}
 		expireOk := seq == lseq && mb.llseq == seq
 
-		if r := sl.MatchWithResult(fsm.subj, &result); len(r.psubs) > 0 {
+		if sl.HasInterest(sm.subj) {
```

Comment:
@neilalexander Note that it used to be `fsm.subj` and you replaced with `sm.subj`. Is it intentional? Are they the same?

Reply:
Good spot, have switched back to `fsm.subj`.